### DISCUSSION TITLE
[MU 4 Issue] Element positioning using arrow keys: 0.5 sp steps or 0.1 sp steps?

### DISCUSSION THREAD

**Discussion by SteveBlower** [musescore/MuseScore#12881] [Ideas]
In MS 4 the arrow keys shift an element by 0.5 sp steps.  0.5 sp is an obvious step size for notes but seems too coarse for fine adjustment, especially of things like slur endpoints and intermediate handles.  In MS 3 the arrow keys shifted elements other than notes in 0.1 sp steps.  

I understand from @MarcSabatella here [https://musescore.org/en/node/327254](https://musescore.org/en/node/327254) that this may have been a design decision.  

Could this step size be made a preference?

Could there be an alternative "fine control" shortcut e.g. [Modifier]+Arrow would move a selected element or handle 0.1sp?   I am not sure what [Modifiers] i.e. combinations of SHIFT, CTRL, ALT with Arrow keys remain available for assignment to such a use.

Could the step size just revert to the MS 3 value of 0.1sp?

**Comment by frfancha** [comment]
Making the step size a preference is the obvious good solution.

**Comment by MarcSabatella** [comment]
FWIW, if I wasn't biased from towards 0.1 from using previous versions of MuseScore, I'd probably say 0.25 sp was the most musically-useful unit.  But I'd also observe the Inspector allows for additional precision regardless of what the keyboard alone does.

**Reply by SnyBr** [reply]
An idea... how about the default is set to 0.25sp. And then OP's suggested "fine-control" shortcut + arrow-keys makes the increments 0.125sp, which is close enough to MS3's 0.1sp, but also allowing for precise positioning at the 0.25sp and 0.5sp positions between staff lines. And then there could be a "bigger-increments" shortcut + arrow keys which changes the increments to 0.5sp or even 1sp.

**Reply by SteveBlower** [reply]
The inspector in MS3 allows for additional precision because the offset spinner works in 0.1 sp steps.  However, in MS4 the offset spinner works in 0.5sp steps.  Yes, you can type in a precise value but that requires you to take note of the current value, do some arithmetic then  type in a new value.  The chances are the value you type in will not achieve exactly what you want and so will need to repeat the process until you get the desired appearance.

I also note that the offset boxes in the appearance tab work on a whole slur, even if only one of the handles is selected.  You can't use them to tweak individual endpoints or the internal handles that adjust curvature.  But that is probably a separate issue.

At present, I think MS4 has a number of shortcomings in the area of adjusting positions - see #12842, #12843, #12844 for example

**Comment by SnyBr** [comment]
I'm all for making the step-size a configurable preference. 0.5sp steps are way too big for any precision-positioning IMO. The properties panel can serve that purpose, but the arrow keys are just so much more convenient.
